Sources/TuistDependencies/SwiftPackageManager/SwiftPackageManagerInteractor.swift

```diff
@@ -0,0 +1,190 @@
+import TSCBasic
+import TuistGraph
+import TuistSupport
+
+enum SwiftPackageManagerInteractorError: FatalError, Equatable {
+    /// Thrown when `Package.resolved` cannont be found in temporary directory after `Swift Package Manager` installation.
+    case packageResolvedNotFound
+    /// Thrown when `.build` directory cannont be found in temporary directory after `Swift Package Manager` installation.
+    case buildDirectoryNotFound
+
+    /// Error type.
+    var type: ErrorType {
+        switch self {
+        case .packageResolvedNotFound,
+             .buildDirectoryNotFound:
+            return .abort
+        }
+    }
+
+    /// Error description.
+    var description: String {
+        switch self {
+        case .packageResolvedNotFound:
+            return "Package.resolved file was not found after Carthage installation."
+        case .buildDirectoryNotFound:
+            return ".build directory was not found after Carthage installation."
+        }
+    }
+}
+
+// MARK: - Swift Package Manager Interacting
+
+public protocol SwiftPackageManagerInteracting {
+    /// Fetches `Swift Package Manager` dependencies.
+    /// - Parameter dependenciesDirectory: The path to the directory that contains the `Tuist/Dependencies/` directory.
+    /// - Parameter dependencies: List of dependencies to intall using `Swift Package Manager`.
+    func fetch(dependenciesDirectory: AbsolutePath, dependencies: SwiftPackageManagerDependencies) throws
+}
+
+// MARK: - Swift Package Manager Interactor
+
+public final class SwiftPackageManagerInteractor: SwiftPackageManagerInteracting {
+    private let fileHandler: FileHandling
+
+    public init(
+        fileHandler: FileHandling = FileHandler.shared
+    ) {
+        self.fileHandler = fileHandler
+    }
+
+    public func fetch(dependenciesDirectory: AbsolutePath, dependencies: SwiftPackageManagerDependencies) throws {
+        logger.info("We are starting to fetch the Swift Package Manager dependencies.", metadata: .section)
+
+        try fileHandler.inTemporaryDirectory { temporaryDirectoryPath in
+            // prepare paths
+            let pathsProvider = SwiftPackageManagerPathsProvider(
+                dependenciesDirectory: dependenciesDirectory,
+                temporaryDirectoryPath: temporaryDirectoryPath
+            )
+
+            // prepare for installation
+            try prepareForInstallation(pathsProvider: pathsProvider, dependencies: dependencies)
+
+            // build command
+            let command = buildCommand(packagePath: temporaryDirectoryPath)
```

Comment:
I'm not sure we need a method for this if the only thing it does is concatenating a bunch of arguments. I'd rather inline the implementation here.